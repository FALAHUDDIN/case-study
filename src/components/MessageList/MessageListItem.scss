.MessageListItem {
  cursor: pointer;

  &--read {
    color: #ddd;
  }
  &--active {
    background: rgba(255, 255, 255, 0.2);
  }
}