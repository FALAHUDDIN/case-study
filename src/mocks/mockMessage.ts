import {IMessage} from "../types/IMessage";

export const MockMessage: IMessage = {
    id: 1,
    isRead: false,
    avatar: "person1.svg",
    sender: "Falah",
    message: "testmessage",
    received: "01/01/2020",
    subject: "testsubject",
    isArchived: false
}
export const MockMessageRead: IMessage = {
    ...MockMessage,
    isRead: true
}
export const MockMessageArchived: IMessage = {
    ...MockMessage,
    isArchived: true
}