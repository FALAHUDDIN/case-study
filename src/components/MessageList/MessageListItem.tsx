import {Box} from "@material-ui/core";
import React, {MouseEventHandler} from "react";
import {IMessage} from "../../types/IMessage"
import FiberNewIcon from '@material-ui/icons/FiberNew';
import './MessageListItem.scss';

export interface Props {
    message: IMessage;
    isActive?: boolean;
    onClick: MouseEventHandler;
}

export const MessageListItem: React.FC<Props> = (props: Props) => {
    const classNames = ['MessageListItem'];
    if (props.isActive) {
        classNames.push('MessageListItem--active');
    }
    if (props.message.isRead) {
        classNames.push('MessageListItem--read');
    }

    return (
        <Box p={1} onClick={props.onClick} className={classNames.join(' ')}>
            <div><strong>{props.message.subject}</strong> {!props.message.isRead && <FiberNewIcon/>}</div>
            <div>{props.message.received}</div>
        </Box>
    )
}

export default MessageListItem