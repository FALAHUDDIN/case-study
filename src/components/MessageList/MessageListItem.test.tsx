import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import {MockMessage, MockMessageRead} from "../../mocks/mockMessage";
import MessageListItem, {Props} from "./MessageListItem";

describe('<MessageListItem/>', () => {
    let wrapper: ShallowWrapper;
    let props: Props = {
        message: MockMessage,
        isActive: true,
        onClick: jest.fn(),
    };

    describe('when active', () => {
        beforeEach(() => {
            const mergedProps = {...props, isActive: true}
            wrapper = shallow(<MessageListItem {...mergedProps} />)
        })

        it('match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        })

        it('has no read class', () => {
            expect(wrapper.find('.MessageListItem').hasClass('MessageListItem--read')).toBeFalsy()
        })

        it('has active class', () => {
            expect(wrapper.find('.MessageListItem').hasClass('MessageListItem--active')).toBeTruthy()
        })
    })

    describe('when inactive', () => {
        beforeEach(() => {
            const mergedProps = {...props, isActive: false}
            wrapper = shallow(<MessageListItem {...mergedProps} />)
        })
    })

    describe('when message is read', () => {
        beforeEach(() => {
            const mergedProps = {...props, message: MockMessageRead}
            wrapper = shallow(<MessageListItem {...mergedProps} />)
        })

        it('has read class', () => {
            expect(wrapper.find('.MessageListItem').hasClass('MessageListItem--read')).toBeTruthy()
        })
    })
});