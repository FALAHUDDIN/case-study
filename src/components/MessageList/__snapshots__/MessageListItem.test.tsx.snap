// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageListItem/> when active match snapshot 1`] = `
<Styled(MuiBox)
  className="MessageListItem MessageListItem--active"
  onClick={[MockFunction]}
  p={1}
>
  <div>
    <strong>
      testsubject
    </strong>
     
    <FiberNewIcon />
  </div>
  <div>
    01/01/2020
  </div>
</Styled(MuiBox)>
`;
